tests/spec/helpers.py

```diff
@@ -0,0 +1,3 @@
+def transpose(results):
```

Comment:
This could do with a docstring, couldn't it.

I understand the confusion -- I'm using _transpose_ to mean switching the rows and columns of a grid.

Our results start as a list of dicts, where each dict is a row.  After transposing, we have a dict of dicts, where the keys of the outer dict are column names, and the values are dicts mapping patients to values in that column.

The value of doing it this way is that we can check each variable in a dataset separately.

However, the tests might be easier to read if take the approach of `test_database.py`, with something like:

```
patients = create_patient_table(
    """
    p |  i
    --+----
    1 | 101
    2 |
    """
)

dataset = Dataset()
dataset.set_population(True)
dataset.v1 = p.i + 1
dataset.v2 = 1 + p.i

assert dataset == parse_dataset(
    """
    p |  v1 | v2
    --+-----+----
    1 | 102 | 102
    2 |     |
    """
)
```

Reply:
Your rationale makes sense: once we start thinking about rows and columns, then _transpose_ is a good choice of verb (it's also consistent with [`pandas.DataFrame.transpose`][1], which we could use as the basis of the docstring). Vega has  a [nest transform][2], which I think was the source of my confusion.

The approach taken in `test_database.py` is easier to read, IMHO. I'm unsure whether it's easier to write; that depends on how forgiving the parser is, in `Table.parse` and `Column.parse` 🙂 

[1]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.transpose.html#pandas.DataFrame.transpose
[2]: https://vega.github.io/vega/docs/transforms/nest/